config/fi_check_package.m4

```diff
@@ -61,7 +61,7 @@ AC_DEFUN([_FI_CHECK_PACKAGE_HEADER], [
 
 
 dnl _FI_CHECK_PACKAGE_LIB(prefix, library, function, extra-libraries,
-dnl                         dir-prefix, libdir,
+dnl                         dir-prefix, libdir, libdir64,
```

Comment:
I think that this approach is a little more complicated than is necessary -- why pass this value in? (which then forces you to update all the components that (directly and indirectly) call this macro)

You know that it is always going to be `$libdir` or `${libdir}64`, so you can wholly hide the extra search for `lib64` in the `_FI_CHECK_PACKAGE_LIB` macro).  Is there a reason you *wouldn't* want to check for `${libdir}` and `${libdir}64`?

In Open MPI, we treat this kind of situation thusly:
* If the user specifies a libdir, use exactly that.  If it fails, then the user specified it wrong (and it's therefore ok to fail configure).
* If the user does not specify a libdir, then check `top_package_dir/lib64` and `top_package_dir/lib` (in that order).  Use the first one that works.

Hence, if an explicit specification is provided, use exactly that.  If only a prefix is specified, then try the `lib64` and `lib` in that prefix and use the first one that works.

Reply:
@jsquyres @j-xiong Thanks for doing review.
Unfortunately, I don't know that this approach is OK (like OpenMPI uses) to check `lib64` for the first time and `lib` after that.
Since  It works for me and you completely agree to do this, I'm going to re-implement in this way.

Reply:
@gladkovdmitry17 You're right that checking `lib64` before `lib` is a heuristic -- there are unlikely-but-possible cases where that isn't the Right thing to do.  But it should likely be "good enough".

Reply:
@jsquyres Thanks for the explanation.
In unlikely-but-possible cases, it'd possible to just overwrite `LDFLAGS` and it should work :)